staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/micro_time_test.go

```diff
@@ -253,3 +253,37 @@ func TestMicroTimeIsZero(t *testing.T) {
 		})
 	}
 }
+
+func TestTimeUnmarshalJSONAndProtoEqual(t *testing.T) {
+	cases := []struct {
+		name   string
+		input  MicroTime
+		result bool
+	}{
+		{"nanosecond level precision", UnixMicro(123, 123123123), true},
+		{"microsecond level precision", UnixMicro(123, 123123000), true},
+	}
```

Comment:
This does a good job of ensuring that round-tripping through protobuf and json marshal → unmarshal ends up with identical data

We also need a test that exercises truncation to nanoseconds during proto unmarshaling... you may need to capture the proto bytes we would serialize a nanosecond-precise microtime to prior to this PR, and embed them in a literal `[]byte{...}` that you pass to a test and ensure they result in a microsecond-precise proto object in memory

Reply:
+1 this doesn't test truncation on unmarshall in its current form.